cpp/include/cudf/detail/gather.hpp

```diff
@@ -11,8 +11,8 @@ namespace cudf {
 namespace experimental {
 namespace detail {
 
-enum class bounds : bool { CHECK, NO_CHECK };
-enum class out_of_bounds : bool { IGNORE, DONT_IGNORE };
+enum class out_of_bounds_policy : int8_t { FAIL, NULLIFY, IGNORE };
```

Comment:
This is great. But now not calling the other one `negative_indices_policy` seems strange.

Reply:
Will update :)

Reply:
Or 

```
enum class index_policy : bool { ALLOW_NEGATIVE, DISALLOW_NEGATIVE };
```

If someone adds another index policy that has nothing to do with negative indices later, we can change it then?

Reply:
Done